src/components/CollectionDetails/index.tsx

```diff
@@ -51,6 +54,22 @@ const CollectionDetails = ({ collection }: CollectionDetailsProps) => {
   const { data: genres } =
     useSWR<{ id: number; name: string }[]>(`/api/v1/genres/movie`);
 
+  useEffect(() => {
+    const tempArray: DownloadingItem[] = [];
+
+    data?.parts.map((item) =>
+      (item.mediaInfo?.downloadStatus4k ?? []).length > 0
+        ? item.mediaInfo?.downloadStatus4k?.forEach((status) =>
+            tempArray.push(status)
+          )
+        : item.mediaInfo?.downloadStatus?.forEach((status) =>
+            tempArray.push(status)
+          )
+    );
+
+    setDownloadStatus(tempArray);
+  }, [data?.parts]);
```

Comment:
We don't need a useEffect or a separate state to do this. you can instead just set a const and use `useMemo` with the data.parts as a dependency.